### PR TITLE
[Jecoute] Added new fields with custom validations

### PR DESCRIPTION
**Pull request by thomasbisignani**

### CODE REVIEW

File: src/Validator/DataSurveyEmailAddressValidator.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class DataSurveyEmailAddressValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof DataSurveyEmailAddress) {
+            throw new UnexpectedTypeException($constraint, DataSurveyEmailAddress::class);
+        }
+
+        if (!$value && $this->context->getObject()->getAgreedToStayInContact()) {
```
Comment (JeremieSamson) [inline src/Validator/DataSurveyEmailAddressValidator.php]: can 'getObject' return null ?

File: translations/validators.fr.yml [hunk 2]
```diff
@@ -163,6 +163,12 @@ survey.answer.unique_choice.with_text_field.invalid: La question {{ surveyQuesti
 survey.answer.multiple_choice.invalid: La question {{ surveyQuestionId }} est une question à choix multiples et doit contenir un ou plusieurs choix sélectionnés.
 survey.answer.multiple_choice.with_text_field.invalid: La question {{ surveyQuestionId }} est une question à choix multiples et ne doit pas contenir de champ texte.
 
+data.survey.gender_choice_other.not_selected: Vous devez sélectionner le choix Autre ou ne pas saisir d'autre genre.
```
Comment (JeremieSamson) [inline translations/validators.fr.yml]: "sélectionner"

File: src/Entity/Jecoute/DataSurvey.php [hunk 3]
```diff
@@ -146,6 +186,56 @@ public function setAgreedToStayInContact(bool $agreedToStayInContact): void
...
+    {
+        return $this->agreedToJoin;
+    }
+
+    public function setAgreedToJoin(bool $agreedToJoin): void
+    {
+        $this->agreedToJoin = $agreedToJoin;
+    }
+
+    public function getPostalCode(): ?int
```
Comment (ottaviano) [inline src/Entity/Jecoute/DataSurvey.php]: if `int`, then `06500` it will be stored as `6500` no?

File: src/Form/Jecoute/DataSurveyFormType.php [hunk 4]
```diff
@@ -47,7 +51,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'allow_add' => true,
                 'by_reference' => false,
             ])
+            ->add('postalCode', IntegerType::class)
```
Comment (ottaviano) [inline src/Form/Jecoute/DataSurveyFormType.php]: i think TextType will be more correct type

**Pull request merged**